gprofiler/main.py

```diff
@@ -557,6 +490,28 @@ def parse_cmd_args():
     return args
 
 
+def _add_profilers_arguments(parser):
+    profilers_registry = get_profilers_registry()
+    for profiler_name, profiler_config in profilers_registry.items():
+        profiler_argument_group = parser.add_argument_group(profiler_name)
+        profiler_argument_group.add_argument(
+            f"--{profiler_name.lower()}-mode",
+            dest=f"{profiler_name.lower()}_mode",
+            default=profiler_config.default_mode,
+            help=profiler_config.profiler_mode_help,
+            choices=profiler_config.possible_modes,
+        )
+        profiler_argument_group.add_argument(
+            f"--no-{profiler_name.lower()}",
+            dest=profiler_name.lower(),
+            action="store_false",
+            default=True,
+            help=f"Disable the profiling of {profiler_name} processes",
+        )
+        for profiler_argument in profiler_config.profiler_arguments:
```

Comment:
I think it's nicer to avoid creating the `profiler_arguments` abstraction and simply allow each registered profiler provide its own `add_my_arguments` method, which you can call. It will get the `parser` object (or actually, the `profiler_argument_group`)

Reply:
Meh, I don't like that at all. I'm trying to make creating a new profiler / adding arguments the easiest process as possible, your solution is too low-level imho

Reply:
Agree. Currently way looks better, just make the `ProfilerArgument` class simpler & this function was already improved for readability with shorter names

Reply:
Simplified the `get_dict` method, really not sure how to shorten the names as they are the same as the kwargs for the `add_argument` method (and they are already very short)

Reply:
By shorted names I referred mostly to `profiler_argument`, `profiler_config` etc